pkg/csi/service/vanilla/controller_helper.go

```diff
@@ -295,3 +295,36 @@ func GetVolumeManagerFromVCHost(ctx context.Context, managers *common.Managers,
 	}
 	return volumeMgr, nil
 }
+
+// getCandidateDSInTopologyForFileVolumes returns a list of shared accessible file service
+// enabled datastores for requested topology.
+func getCandidateDSInTopologyForFileVolumes(ctx context.Context,
+	topologyArr []*csi.Topology, vcenter *vsphere.VirtualCenter) ([]*vsphere.DatastoreInfo,
+	error) {
+	log := logger.GetLogger(ctx)
+
+	var sharedDatastores []*vsphere.DatastoreInfo
+	// A topology requirement is an array of topology segments.
+	for _, topology := range topologyArr {
+		segments := topology.GetSegments()
+		hostMoRefs, err := common.GetHostsForSegment(ctx, segments, vcenter)
+		if err != nil {
+			return nil, logger.LogNewErrorCodef(log, codes.Internal,
+				"failed to fetch hosts belonging to topology %+v. Error: %+v", topologyArr, err)
+		}
+		sharedDatastoresForHosts, err := vsphere.GetSharedDatastoresForHosts(ctx, hostMoRefs)
```

Comment:
we do not need to get shared datastores for file volumes. We just need to find the datastores that these hosts can access. This can be done using function https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/common/cns-lib/vsphere/hostsystem.go#L43

Reply:
Isn't `GetSharedDatastoresForHosts` just a wrapper for `GetAllAccessibleDatastores` ? We can reuse this method, right ?

Reply:
No, it is not. We do not care if a datastore is shared or not among the given hosts. We just want all the datastores.

Reply:
Okay, got it. Thanks! I have addressed it in the latest commit. Resolving the comment